src/pages/verify/guides/privatekey-jwt.mdx

```diff
@@ -0,0 +1,160 @@
+
+#### Example
+
+In the code sample below, the assertion is generated using the [jose library](https://github.com/panva/jose):
+
+```javascript
+const { SignJWT } = require('jose');
+
+const signingKey = 'YOUR_SIGNING_KEY_HERE';
+
+async function generateJWT(signingKey) {
+  try {
+    const jwt = await new SignJWT({})
+      .setProtectedHeader({ alg: 'RS256' })
+      .setIssuedAt()
+      .setIssuer('CLIENT_ID')
+      .setSubject('CLIENT_ID')
+      .setAudience('YOUR_CRIIPTO_DOMAIN')
+      .setExpirationTime('5m')
+      .sign(signingKey);
+    console.log('JWT:', jwt);
+    return jwt;
+  } catch (error) {
+    console.error('Error generating JWT:', error.message);
+    throw error;
+  }
+}
+```
+
+### Exchange the client assertion for an access token
+
+You'll need to include the client assertion along with the verification code when performing the [code-for-token exchange](/verify/getting-started/oidc-intro/#exchange-the-code-for-a-token).
+
+The `client_assertion_type` should be set to `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`.
+
+```curl
+# Replace `$client_assertion` with your JWT assertion
+# Replace `CLIENT_ID` with your Client ID
+# Replace `AUTHORIZATION_CODE` with the authorization code received in the initial authorization request
+# Replace `YOUR_DOMAIN.criipto.id` with your Criipto domain
+# Replace `YOUR_RETURN_URL` with the redirect URI used in the initial authorization request
+
+HTTP POST https://YOUR_DOMAIN.criipto.id/oauth2/token
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=authorization_code
+&client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer
+&client_assertion=$client_assertion
+&code=AUTHORIZATION_CODE
+&client_id=CLIENT_ID
+&redirect_uri=YOUR_RETURN_URL
+```
+
+## Configure Private Key JWT Authentication
+
+To configure Private Key JWT Authentication, you'll need to register JWKS on behalf of your client on the [management dashboard](https://dashboard.criipto.com).
+
+In your Application settings, go to `OpenID Connect` > `Client JWKS`. You can add a static JWKS or a JWKS endpoint.
+
+![Client JWKS Endpoint](./images/JWKS_endpoint.png)
+
+- Static: Add your static JWKS.
+- JWKS Endpoint: Provide JWKS URI to fetch JWKS dynamically.
+
+## Advantages of Private Key JWT Authentication
+
+`private_key_jwt` offers enhanced security compared to traditional `client_secret` authentication methods by removing the need for a pre-shared symmetric key (client secret).
+
+Instead, you generate a pair of asymmetric keys and have Criipto point to the JWKS in your system. During authentication, your application signs its request with the private key, creating a signed client assertion. Criipto Verify can then use the public key from the JWKS to decrypt and verify the assertion.
+
+This setup also facilitates seamless key rollover.
```

Comment:
I think "This setup" might be interpreted as `private_key_jwt`, but the seamless key rollover claim is only true when using the JWKS endpoint, not with the static URI.
Alternative off-the-top-of-my-head phrasing:
"When using `private_key_jwt` with a JWKS endpoint as opposed to a static JWKS you can benefit from seamless key rollover effectively avoiding change anything in the Criipto dashboard if you chose to rotate your keys."
Adjust appropriately if you find this helpful :)

Reply:
Very helpful. Thanks a lot @kasperhj. It's updated now.